compiler/src/dotty/tools/dotc/transform/TreeTransform.scala

```diff
@@ -209,15 +211,29 @@ object TreeTransforms {
    *  @see NXTransformations.index for format of plan
    */
   class NXTransformations {
+    private val clsMethodsCache = new java.util.IdentityHashMap[Class[_], Array[java.lang.reflect.Method]]
 
+    // TODO: We spend too much time here. See if we can call it less or make it faster,
+    // e.g. by checking `cls.getMethod(name, ...).getDeclaringClass != classOf[TreeTransform]` instead.
     private def hasRedefinedMethod(cls: Class[_], name: String): Boolean = {
-      val clsMethods = cls.getDeclaredMethods
+      if (cls.eq(classOf[TreeTransform]) || cls.eq(classOf[NoTreeTransform]) ||
```

Comment:
Not clear we need this anymore, because of ed5eb24.

We have just reduced the # of calls to this method by a factor 7.

Reply:
Yeah, it's more of a micro-optimization now. I can remove the added cache map if you think it's not worth the complexity now.